pilz_robot_programming/src/pilz_robot_programming/robot.py

```diff
@@ -237,7 +240,11 @@ def stop(self):
         """
         rospy.loginfo("Stop called.")
         self._move_ctrl_sm.switch(MoveControlAction.STOP)
-        self._cancel_on_all_clients()
+
+        with self._move_ctrl_sm: # wait, if _execute is just starting a send_goal()
+            actionclient_state = self._sequence_client.get_state()
+            if any(state == actionclient_state for state in _VALID_GOAL_STATUS_FOR_CANCEL):
```

Comment:
Wouldn't `if actionclient_state in _VALID_GOAL_STATUS_FOR_CANCEL` work?

Reply:
It works.